File: app/src/main/java/ru/prsolution/winstrike/WinstrikeApp.java
package ru.prsolution.winstrike;

import android.app.Application;
import android.util.DisplayMetrics;

import com.crashlytics.android.Crashlytics;

import java.io.File;

import io.fabric.sdk.android.Fabric;
import retrofit2.Retrofit;
import ru.prsolution.winstrike.db.entity.UserEntity;
import ru.prsolution.winstrike.di.AppComponent;
import ru.prsolution.winstrike.di.DaggerAppComponent;
import ru.prsolution.winstrike.di.module.ContextModule;
import ru.prsolution.winstrike.di.module.NetworkModule;
import ru.prsolution.winstrike.mvp.apimodels.RoomLayout;
import ru.prsolution.winstrike.mvp.models.SeatModel;
import ru.prsolution.winstrike.networking.Service;


public class WinstrikeApp extends Application {
    public static final boolean DEBUG = false;
    public static WinstrikeApp INSTANCE;
    private AppComponent sAppComponent;
    private UserEntity user;
    private SeatModel seat;
    private RoomLayout roomLayout;

    @Override
    public void onCreate() {
        super.onCreate();
        Fabric.with(this, new Crashlytics());
        INSTANCE = this;
    }

    public static WinstrikeApp getInstance() {
        return INSTANCE;
    }


    public AppComponent getAppComponent() {
        if (sAppComponent == null) {
            File cacheFile = new File(getCacheDir(), "responses");
            sAppComponent = DaggerAppComponent.builder()
                    .contextModule(new ContextModule(this))
                    .networkModule(new NetworkModule(cacheFile))
                    .build();
        }
        return sAppComponent;
    }

    public Service getService() {
        File cacheFile = new File(getCacheDir(), "responses");
        NetworkModule networkModule = new NetworkModule(cacheFile);
        Retrofit retrofit = networkModule.provideCall();
        return new Service(networkModule.providesNetworkService(retrofit));
    }


    public float getDisplayWidhtDp() {
        DisplayMetrics displayMetrics = this.getResources().getDisplayMetrics();
        float dpWidth = displayMetrics.widthPixels / displayMetrics.density;
        return dpWidth;
    }

    public float getDisplayHeightDp() {
        DisplayMetrics displayMetrics = this.getResources().getDisplayMetrics();
        float dpHeight = displayMetrics.heightPixels / displayMetrics.density;
        return dpHeight;
    }

    public float getDisplayHeightPx() {
        DisplayMetrics displayMetrics = this.getResources().getDisplayMetrics();
        float dpHeight = displayMetrics.heightPixels;
        return dpHeight;
    }

    public float getDisplayWidhtPx() {
        DisplayMetrics displayMetrics = this.getResources().getDisplayMetrics();
        float dpWidth = displayMetrics.widthPixels;
        return dpWidth;
    }

    public UserEntity getUser() {
        return this.user;
    }

    public void setSeat(SeatModel seat) {
        this.seat = seat;
    }

    public SeatModel getSeat() {
        return seat;
    }

    public RoomLayout getRoomLayout() {
        return this.roomLayout;
    }

    public void setRoomLayout(RoomLayout roomLayout) {
        this.roomLayout = roomLayout;
    }
}

File: app/src/main/kotlin/ru/prsolution/winstrike/WinstrikeApp.kt
package ru.prsolution.winstrike

import android.app.Application
import com.facebook.drawee.backends.pipeline.Fresco

import ru.prsolution.winstrike.datasource.model.ArenaEntity
import ru.prsolution.winstrike.datasource.model.RoomEntity
import ru.prsolution.winstrike.domain.models.SeatCarousel
import ru.prsolution.winstrike.domain.models.login.UserEntity
import ru.prsolution.winstrike.presentation.utils.pref.PrefUtils


class WinstrikeApp : Application() {
	val user: UserEntity? = null
	var seat: SeatCarousel? = null
	var roomLayout: RoomEntity? = null
	var rooms: List<ArenaEntity>? = null




	val displayWidhtDp: Float
		get() {
			val displayMetrics = this.resources.displayMetrics
			val dpWidth = displayMetrics.widthPixels / displayMetrics.density
			return dpWidth
		}

	val displayHeightDp: Float
		get() {
			val displayMetrics = this.resources.displayMetrics
			val dpHeight = displayMetrics.heightPixels / displayMetrics.density
			return dpHeight
		}

	val displayHeightPx: Float
		get() {
			val displayMetrics = this.resources.displayMetrics
			val dpHeight = displayMetrics.heightPixels.toFloat()
			return dpHeight
		}

	val displayWidhtPx: Float
		get() {
			val displayMetrics = this.resources.displayMetrics
			val dpWidth = displayMetrics.widthPixels.toFloat()
			return dpWidth
		}

	override fun onCreate() {
		super.onCreate()
		//        Fabric.with(this, new Crashlytics());
		instance = this

		Fresco.initialize(this)

		initScreenPref()

	}

	private fun initScreenPref() {
		PrefUtils.displayHeightPx = displayHeightPx
		PrefUtils.displayWidhtPx = displayWidhtPx
		PrefUtils.displayHeightDp = displayHeightDp
		PrefUtils.displayWidhtDp = displayWidhtDp
	}

	companion object {
		lateinit var instance: WinstrikeApp
	}
}
